src/components/Textfields/IconTextField/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { InputAdornment } from "@material-ui/core";
+import TextFieldBase, { TextfieldBaseProps } from "src/components/Textfields/TextFieldBase";
+
+export interface IconTextFieldProps extends TextfieldBaseProps {
+    icon?: React.ReactNode;
+}
+
+const IconTextField = (props: IconTextFieldProps) => {
```

Comment:
I'm not sure if this should be made into a component, this should be a story of textfieldbase?